### PR TITLE
feat(elements|ino-carousel): add `valueChange` event to inform about slide changes on arrow click

### PR DESCRIPTION
**Pull request by TobiasHeimGalindo**
As the value changes upon clicking the directional icon buttons the current slide value gets emitted.

### CODE REVIEW

File: packages/elements/src/components/ino-carousel/ino-carousel.tsx [hunk 1]
```diff
@@ -144,6 +153,31 @@ export class InoCarousel implements ComponentInterface {
     }
   };
 
+  private emitSlideChange(iconArrow: ArrowDirections): void {
+    const carouselSlide = this.getSlides()[this.currentSlide];
+    const allSlides = this.getSlides();
+
+    if (!carouselSlide) {
+      this.valueChange.emit(allSlides[0].value);
```
Comment (BenPag) [inline packages/elements/src/components/ino-carousel/ino-carousel.tsx:161]: Please check if `allSlides[0]` is `null `or `undefined`

File: packages/elements/src/components/ino-carousel/ino-carousel.tsx [hunk 2]
```diff
@@ -144,6 +153,31 @@ export class InoCarousel implements ComponentInterface {
...
+    switch (iconArrow) {
+      case ArrowDirections.RIGHT:
+        nextSlide =
+          (carouselSlide.nextElementSibling as HTMLInoCarouselSlideElement) ??
+          allSlides[0];
+        break;
+      case ArrowDirections.LEFT:
+        nextSlide =
+          (carouselSlide.previousElementSibling as HTMLInoCarouselSlideElement) ??
+          allSlides[allSlides.length - 1];
```
Comment (BenPag) [inline packages/elements/src/components/ino-carousel/ino-carousel.tsx:175]: To be sure, check if the array is not empty.

File: packages/elements/src/components/ino-carousel/ino-carousel.e2e.ts [hunk 3]
```diff
@@ -34,3 +35,62 @@ describe('ino-carousel', () => {
     });
   });
 });
+
+describe('InoRadioGroup', () => {
```
Comment (BenPag) [inline packages/elements/src/components/ino-carousel/ino-carousel.e2e.ts:39]: Naming is incorrect i guess

**Review by BenPag — CHANGES_REQUESTED** [review]

File: packages/elements/src/components/ino-carousel/ino-carousel.tsx [hunk 4]
```diff
@@ -144,6 +153,33 @@ export class InoCarousel implements ComponentInterface {
...
+        nextSlide =
+          (carouselSlide.nextElementSibling as HTMLInoCarouselSlideElement) ??
+          allSlides[0];
+        break;
+      case ArrowDirections.LEFT:
+        if (allSlides.length !== 0) {
+          nextSlide =
+            (carouselSlide.previousElementSibling as HTMLInoCarouselSlideElement) ??
+            allSlides[allSlides.length - 1];
+        }
```
Comment (janivo) [inline packages/elements/src/components/ino-carousel/ino-carousel.tsx:177]: We've confirmed that all slides is not empty
```suggestion
	      nextSlide =
	        (carouselSlide.previousElementSibling as HTMLInoCarouselSlideElement) ??
	        allSlides[allSlides.length - 1];
```

File: packages/elements/src/components/ino-carousel/ino-carousel.tsx [hunk 5]
```diff
@@ -28,13 +34,16 @@ export class InoCarousel implements ComponentInterface {
...
   @Prop() value?: string;
-
   @Watch('value')
   valueChanged(newVal: string) {
     this.addAnimationToSlide(this.slides[this.currentSlide]); // adds the slide animation to the current slide
     this.selectSlide(newVal);
     this.addAnimationToSlide(this.slides[this.currentSlide]); // adds the slide animation to the new slide
   }
+  /**
+   * Emits the value of the slide once it changed
```
Comment (janivo) [inline packages/elements/src/components/ino-carousel/ino-carousel.tsx:44]: The slides have not been changed on arrow press. We inform the consumer that the slide should be changed.
```suggestion
   * Emits the `value` of the slide that should be displayed after the left or right arrow has been clicked.
```

**Review by janivo — CHANGES_REQUESTED** [review]
Nice job overall! 💥 
After you've applied the suggestions, run `yarn build` once again and commit the generated stuff, because we've made changes to the documentation.

File: packages/elements/src/components/ino-carousel/ino-carousel.tsx [hunk 6]
```diff
@@ -154,10 +154,11 @@ export class InoCarousel implements ComponentInterface {
   };
 
   private emitSlideChange(iconArrow: ArrowDirections): void {
-    const carouselSlide = this.getSlides()[this.currentSlide];
     const allSlides = this.getSlides();
+    if (!allSlides || allSlides.length === 0) return;
+    const carouselSlide = this.getSlides()[this.currentSlide];
```
Comment (janivo) [inline packages/elements/src/components/ino-carousel/ino-carousel.tsx:159]:
```suggestion
    const allSlides = this.getSlides();
    
    if (!allSlides || allSlides.length === 0) return;
    
    const carouselSlide = allSlides[this.currentSlide];
```

File: packages/elements/src/components/ino-carousel/ino-carousel.tsx [hunk 7]
```diff
@@ -144,6 +153,33 @@ export class InoCarousel implements ComponentInterface {
     }
   };
 
+  private emitSlideChange(iconArrow: ArrowDirections): void {
+    const allSlides = this.getSlides();
+    if (!allSlides || allSlides.length === 0) return;
+    const carouselSlide = this.getSlides()[this.currentSlide];
```
Comment (janivo) [inline packages/elements/src/components/ino-carousel/ino-carousel.tsx:159]:
```suggestion
    const allSlides = this.getSlides();
    
    if (!allSlides || allSlides.length === 0) return;
    
    const carouselSlide = this.getSlides()[this.currentSlide];
```

**Pull request merged**